genlayer_py/types/transactions.py

```
@@ -413,17 +414,12 @@ def _decode_input_data(self) -> Union[DecodedDeployData, DecodedCallData, None]:
+                    f"[decode_input_data] Unexpected RLP array length: {len(rlp_decoded_array)} {rlp_decoded_array}",
                 )
                 return None
         except Exception as e:
-            print(
-                "[decode_input_data] Error decoding RLP:",
-                str(e),
-                "Raw RLP App Data:",
-                self.tx_data,
+            logging.warning(
+                f"[decode_input_data] Error decoding RLP: {e} Raw RLP App Data: {self.tx_data}",
             )
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)